extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/reactive/runtime/ReactiveMongoOperations.java

```diff
@@ -383,11 +394,17 @@ String bindUpdate(Class<?> clazz, String query, Object[] params) {
     /**
      * We should have a query like <code>{'firstname': :firstname, 'lastname': :lastname}</code> for native one
      * and like <code>firstname = :firstname and lastname = :lastname</code> for PanacheQL one.
-     * As update document needs a <code>$set</code> operator we add it if needed.
+     * As update document needs an update operator, we add <code>$set</code> if none is provided.
      */
     String bindUpdate(Class<?> clazz, String query, Map<String, Object> params) {
         String bindUpdate = bindQuery(clazz, query, params);
-        if (!bindUpdate.contains("$set")) {
+        boolean containsOperator = false;
```

Comment:
duplicated code could probably be moved to a method

Reply:
I move the code on a dedicated method.
There is still duplication between the reactive and imperative variant but it is consistent with other methods/params that are duplicated between the two.